<launch>
	<!-- Gazebo world -->
	<arg name="world_file"  default="$(find evolutive_map)/worlds/empty.world"/>

	<arg name="base"      value="$(optenv TURTLEBOT_BASE kobuki)"/>
	<arg name="battery"   value="$(optenv TURTLEBOT_BATTERY /proc/acpi/battery/BAT0)"/>
	<arg name="gui" default="true"/>
	<arg name="stacks"    value="$(optenv TURTLEBOT_STACKS hexagons)"/>
	<arg name="3d_sensor" value="kinect"/>  <!-- kinect or hokuyo -->

	<include file="$(find gazebo_ros)/launch/empty_world.launch">
		<arg name="use_sim_time" value="true"/>
		<arg name="debug" value="false"/>
		<arg name="gui" value="$(arg gui)" />
		<arg name="world_name" value="$(arg world_file)"/>
	</include>

	<!-- Turtlebot -->
	<include file="$(find evolutive_map)/launch/includes/kobuki.launch.xml">
		<arg name="base" value="$(arg base)"/>
		<arg name="stacks" value="$(arg stacks)"/>
		<arg name="3d_sensor" value="$(arg 3d_sensor)"/>
	</include>

	<node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
		<param name="publish_frequency" type="double" value="30.0" />
	</node>

	<!-- Fake laser -->
	<node pkg="nodelet" type="nodelet" name="laserscan_nodelet_manager" args="manager"/>
	<node pkg="nodelet" type="nodelet" name="depthimage_to_laserscan"
		args="load depthimage_to_laserscan/DepthImageToLaserScanNodelet laserscan_nodelet_manager">
		<param name="scan_height" value="10"/>
		<param name="output_frame_id" value="/camera_depth_frame"/>
		<param name="range_min" value="0.45"/>
		<remap from="image" to="/camera/depth/image_raw"/>
		<remap from="scan" to="/scan"/>
	</node>

	<!-- life_cycle -->
	<node name="life_cycle" type="life_cycle" pkg="evolutive_map" output="screen">
		<param name="mapImagePath" type="string" value="ressources/aerial_views/ExperimentMap_medium.jpg"/>
		<param name="maplimits" value="40"/>
		<param name="minsleeptime" value="40.0"/> <!-- 60 -->
		<param name="randsleeptime" value="80.0"/> <!-- 120 -->
		<param name="leftoversRatio" value="0.1"/>
		<param name="securitySpawningDistance" value="2.0"/>
	</node>
</launch>
